modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala

```diff
@@ -238,19 +271,38 @@ private[elasticsearch] final class HttpElasticExecutor private (config: ElasticC
         case HttpOk =>
           response.body.fold(
             e => ZIO.fail(new ElasticException(s"Exception occurred: ${e.getMessage}")),
-            value => ZIO.succeed(new SearchResult(value.results.map(Item)))
+            value => ZIO.succeed((Chunk.fromIterable(value.results).map(Item), value.scrollId))
           )
         case _ =>
           ZIO.fail(createElasticExceptionFromCustomResponse(response))
       }
     }
 
-  private def executeGetByQueryWithScroll(r: Search): Task[(Chunk[Item], Option[String])] =
+  private def executeGetByScroll(scrollId: String): Task[(Chunk[Item], Option[String])] =
     sendRequestWithCustomResponse(
       request
-        .post(
-          uri"${config.uri}/${r.index}/$Search".withParams((Scroll, ScrollDefaultDuration))
-        )
+        .post(uri"${config.uri}/$Search/$Scroll".withParams((Scroll, ScrollDefaultDuration)))
+        .response(asJson[ElasticQueryResponse])
+        .contentType(ApplicationJson)
+        .body(Obj(ScrollId -> Str(scrollId)))
+    ).flatMap { response =>
+      response.code match {
+        case HttpOk =>
+          response.body.fold(
+            e => ZIO.fail(new ElasticException(s"Exception occurred: ${e.getMessage}")),
```

Comment:
Put `ZIO.fail(new ElasticException(s"Exception occurred: ${e.getMessage}"))` to new line.